cli/signals.go

```diff
@@ -54,13 +57,27 @@ func backtrace() {
 }
 
 func fatalSignal(sig syscall.Signal) bool {
-	return sigFatal[sig]
+	_, exists := handledSignalsMap[sig]
```

Comment:
Why are you checking the same map `handledSignalsMap` from both `fatalSignal()` and `nonFatalSignal()`. Am I missing something ?

Reply:
That map contains all the signals that are handled. But `SIGUSR1` is handled differently - the bool map value is `false` meaning "don't die on receipt of this signal". The two functions you mention are similar but not that `nonFatalSignal()` is only dealing with `SIGUSR1` and hence has the `!handledSignalsMap[sig]` test at the end.

Reply:
Oh yes, I missed this slight difference !